bundle/edu.gemini.shared.util/src/main/java/edu/gemini/shared/util/immutable/Either.java

```diff
@@ -0,0 +1,80 @@
+package edu.gemini.shared.util.immutable;
+
+import java.io.Serializable;
+
+/**
+ * Simulates a Scala Either[S,T].
+ */
+public final class Either<L,R> implements Serializable {
```

Comment:
Let's make it `ImEither` (like `ImOption`) to avoid confusion with scala `Either`?